index.yaml

```diff
@@ -363,6 +363,48 @@ entries:
     - assets/fleet-crd/fleet-crd-0.3.000.tgz
```

Comment:
> The index.yaml will still point longhorn 100.0.0+up1.1.2 at assets/longhorn/longhorn-100.0.0+up1.1.2.tgz instead of assets/longhorn-1.1/longhorn-100.0.0+up1.1.2.tgz.
> This is expected since that version is already released; however, the first time that longhorn pushes a change into the 1.1.x line, we should delete that orphaned chart.

We need to add a rancher-max of `2.6.0` to 100.0.0+up1.1.2, @aiyengar2 does that require a re-release of the chart or are we just going to modify it on the release branch ?

Reply:
We have to modify it in the release branch; PR workflows won't pass in dev-v2.6 if we're trying to modify an already released chart.

Reply:
Ok, that can be handled as part of this: https://github.com/rancher/rancher/issues/34479#issuecomment-918619403

@innobead here we just need to add `rancher-version` annotation on both of these tars being introduced:
* 1.1.2 should be min of 2.6.0 and max of 2.6.1 (the max because of the upgrade-to-older-version problem)
* 1.2.0 should be min of 2.6.1 and no max

See linked issue above for more info.

Reply:
@cbron could u help explain what the upgrade-to-older-version problem here? Because it looks u wants to have max version for 1.1.2 as 2.6.1, does that means users can’t install 1.1.2 after 2.6.1? If that’s the case, then why?

Reply:
Because of our 100 versioning scheme. If we introduce 1.2.0 in 2.5 then a user could "upgrade" from 1.2.0 to 100.0.0+up1.1.2 and actually perform a downgrade from 1.2.0->1.1.2. If we limit what is available with min/max and the users go to the latest rancher 2.6.1, then the "older" version won't be available and they would stay on 1.2.0. We also try to help people install the latest version and not older ones.

Reply:
Got your point for 100 versioning.

But as I mentioned above, right now Longhorn supports multiple releases for different minor releases. (Currently it’s 1.1.x and 1.2.x)

So we should allow users staying in 1.1.x and being able to choose to upgrade to the next patch version (1.1.2) instead of only upgrading to the latest minor release (1.2.0).

Also in the future, we will have 1.2.1 and 1.1.3 in following releases. 

So the question is whether users are able to install or upgrade to 1.1.2 in 2.6.1 if we set max versions for 1.1.2 as u suggested?

Reply:
They would not be able to upgrade to 1.1.2 if we set the max at less than the rancher version. For other charts thats how we handle it, we just move to the latest. If you want to keep both lines open to install/upgrade, then the tradeoff is that they can accidentally downgrade. I suppose longhorn can be an exception here too, perhaps you should warn about it in longhorn docs or release notes.

Reply:
Thanks, so what we really need to add based on our discussion is:

1.1.2/1.2.0 should be min of 2.6.0 and no max, correct?

Also, for longhorn releases starting from 2.6.0, I think we only need to add min of 2.6.0, because the upgrade issue will not happen for charts in 2.6, because the upgrade will rely on the versions we discussed here (https://github.com/rancher/charts/pull/1428#issuecomment-914651849) w/ corresponding minor/patch updates of downstream charts for the upstream charts.

Reply:
> 1.1.2/1.2.0 should be min of 2.6.0 and no max, correct?

If you are ok with the potential downfall of the false-upgrade here then yes. But we shouldn't do it long term. 

>  for longhorn releases starting from 2.6.0, I think we only need to add min of 2.6.0 w/o max, because the upgrade issue will not happen for charts in 2.6 and the chart upgrade will rely on the versions we discussed

Doing this permanently is going to bring up problems, besides the false-upgrade situation. If there are rancher requirements in place that longhorn needs a specific version of rancher to be compatible with, for example which versions of kubernetes rancher does and does not support. Also once we do have the fix for airgap in place we'll have to be more specific about what we allow in airgap, so the size of the total-airgap images doesn't get too large.

Reply:
Agree, no perfect solution if we want support multiple maintenance line releases and also keep releasing in 2.5 and 2.6.

But the proposal is:
1. Keep the latest releases of each release line
2. Disable previous releases of each release line (rerelease or update the released one directly?)

This can avoid false upgrade for each line (ex: 1.1.x at rancher 2.5 to 1.1.<latest> at rancher 2.6 or 1.2.x at rancher 2.5 to 1.2.<latest> at rancher 2.6) but cross line false upgrade can't (ex: 1.2.x at 2.5 to 1.1.x at 2.6). This part needs to tackle by documentation).

WDYT? Any better ideas are welcome.

Reply:
That sounds like a good compromise for now. We should note that in here: https://github.com/rancher/rancher/issues/34589

2.6.1: 
1.1.2-min of 2.6.0
1.2.0-min of 2.6.0 

2.6.2: 
1.1.2-min of 2.6.0, max of 2.6.1
1.2.0-min of 2.6.0, max of 2.6.1
1.1.3-min of 2.6.2
1.2.1-min of 2.6.2

Reply:
The example is what I thought now but if we don’t have any release in 2.6.2 but 2.6.3 instead, the example will be like this: (however the same concept)

2.6.3:
1.1.2-min of 2.6.0, max of 2.6.2
1.2.0-min of 2.6.0, max of 2.6.2
1.1.3-min of 2.6.3
1.2.1-min of 2.6.3

If no other concerns, lets do this way first but also need to share this info with PM to let them know this change introduced by 2.6 introduction, different version scheming between 2.5 n 2.6, multiple releases support and continuous releases to 2.5 n 2.6.

I will update the PR. Thanks @cbron

Reply:
Yes that is correct, we won't add the previous version's max until new versions are going out. We don't have great mechanism to handle that (updating the older versions) right now but should soon.